attic/rfc_metrics.md

```diff
@@ -36,14 +51,55 @@ Internally, the function would look the following:
 
 Example:
 ```r
-fn = function(truth, response, positive, ...) {
-  assert_binary(truth, response = response, positive = positive)
-  fn_cm(cm(truth, response, positive))
-}
+MeasureFairness = R6Class("MeasureFairness", inherit = Measure, cloneable = FALSE,
+  public = list(
+    fun = NULL,
+    na_value = NaN,
+    base_measure = NULL,
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    initialize = function(name, base_measure) {
+      info = mlr3fairness::measures[["groupwise_abs_diff"]]
+      super$initialize(
+        id = paste0("fairness.", base_measure$id),
+        range = c(info$lower, info$upper),
+        minimize = info$minimize,
+        predict_type = info$predict_type,
+        packages = "mlr3fairness",
+        man = paste0("mlr_measures_fairness.", name)
+      )
+      self$fun = get(name, envir = asNamespace("mlr3fairness"), mode = "function")
+      self$base_measure = base_measure
+    }
+  ),
+
+  private = list(
+    .score = function(prediction, task, ...) {
+      invoke(self$fun, prediction = prediction, na_value = self$na_value, data_task = task,
+           base_measure = self$base_measure)
+    }
+  )
+)
+
+mlr_measures$add("fairness.groupwise_abs_diff", MeasureFairness, name = "groupwise_abs_diff")
+```
 
-fn_cm = function(m, na_value = NaN) {
-  m[2L, 1L]
+The groupwise operations are performed inside the Fairness Metric functions. See below:
+```r
+groupwise_abs_diff <- function(prediction, base_measure, positive, data_task, response = NULL, ...){
+  #Assert the status for all the parameters
+
+  subcol = data_task$col_roles$pta
+  prediction = split(as.data.table(prediction), data_task$data(cols = subcol))
```

Comment:
So there is also `PredictionData` which is part of the `Prediction`. 
Can this help us somehow or can we just ignore it?